WordPress/gutenberg
# Remove the lock icon in list views for content only blocks

## What?

Part of https://github.com/WordPress/gutenberg/issues/60834. Part of https://github.com/WordPress/gutenberg/issues/59819.

Remove the lock icon in list views for content only blocks.

~This PR also restore the more menu in the list view as I think there are still some actions useful for content only blocks:~
~1. Copy~
~2. Copy styles~
~3. Paste styles~
~4. Edit as HTML~

~We can decide separately if any of these aren't needed or anything else should be added to the list though, as well as the more menu in the block toolbar.~

The description design exploration shown in https://github.com/WordPress/gutenberg/issues/60834#issuecomment-2063662703 can be implemented as a follow-up of this PR. (PR: #61127)

## Why?


See https://github.com/WordPress/gutenberg/issues/60834. The lock icon normally indicates non-editable status, which is not true for content only blocks. This PR removes the lock icon.

## How?

Add `isContentOnly` check in multiple places

## Testing Instructions




1. Create a synced pattern and add some overridable blocks.
6. Create a post and insert the pattern.
7. The list view should only show the overridable blocks for the pattern.
8. The overridable blocks should not have lock icons.

## Screenshots or screencast 

**Before**:
![image](https://github.com/WordPress/gutenberg/assets/7753001/a8423b4f-7259-45d4-90f9-5ec8ac2c95f2)

**After**:
![image](https://github.com/WordPress/gutenberg/assets/7753001/f3290f0c-6ca6-424d-9d46-cad20cd0e2ab)
